app/controllers/builds_controller.rb

```diff
@@ -36,21 +36,35 @@ def new
   def create
     new = false
     saved = false
+    status = :no_content
+    external_build_has_digest = false
 
     Samson::Retry.retry_when_not_unique do
       if registering_external_build? && @build = find_external_build
-        return head :unprocessable_entity if @build.docker_repo_digest
+        external_build_has_digest = @build.docker_repo_digest.present?
         @build.attributes = edit_build_params(validate: false)
       else
         @build = scope.new(new_build_params.merge(creator: current_user))
       end
 
       new = @build.new_record?
-      saved = @build.save
+      updated = @build.changed?
+      status =
+        if new
+          :created
+        elsif updated
+          :ok
+        else
+          :no_content
```

Comment:
not sure if we even need that ... :ok would work too ?

Reply:
I think you're right. I was under the impression that the client would need unique status codes depending on what happened (update vs no change) but it sounds like that's not the case anymore so just a :ok would work.